### PR TITLE
Replace handmade lint script with official action 

### PR DESCRIPTION
**Pull request by tisonkun**
This refers to #290.

The `helm/chart-releaser-action` update `index.yaml` in an incremental manner, so we need to push the existing `index.yaml` file to `gh-pages` later.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -17,39 +17,36 @@
...
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
 
-      - name: Detect changed files
-        id:   changes
-        uses: apache/pulsar-test-infra/paths-filter@master
+      - name: Install Helm
+        uses: azure/setup-helm@v1
```
Comment (tisonkun) [inline .github/workflows/release.yml:45]: This action is too complex to bundle. An INFRA ticket already adds this action to the whitelist: https://issues.apache.org/jira/browse/INFRA-23716

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -43,9 +42,22 @@ jobs:
...
+      - name: Set up Helm
+        if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+        uses: azure/setup-helm@v3
+        with:
+          version: v3.10.0
+
+      - name: Set up Python
         if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
-        id: lint
-        uses: helm/chart-testing-action@v2.0.0
```
Comment (tisonkun) [inline .github/workflows/lint.yml:49]: This action is never allowed by INFRA settings and thus we always failed before. Fix on the fly.

**Comment by tisonkun** [comment]
cc @michaeljmarshall @ericsyh @mattisonchao @yaalsn

**Review by wolfstudy — APPROVED** [review]
Cool LGTM +1

**Comment by tisonkun** [comment]
@michaeljmarshall Thanks for your follow-up. Let me check your investigation today and feedback. My first object is to decouple the Chart index file from the website :)

**Comment by tisonkun** [comment]
> will break

Yep. As mentioned in #290, we can set a server-side redirection in the website `.htaccess`. The reason is that, website build process doesn't handle Chart yaml index so that it's possible we accidentally remove the file - they're totally different topics.

Setting up a redirection is in the domain of a website maintenance and avoid further development accidentally modifies it.

**Comment by michaeljmarshall** [comment]
Sure, I agree that the current workflow is a bit of a hack, and the current `index.yaml` is not very well protected. I added a comment to #290, too.

**Comment by tisonkun** [comment]
@michaeljmarshall I've narrow the scope to replace the lint scripts with `chart-testing-action`. PTAL.

I'm going through your proposal on release process now.

**Comment by tisonkun** [comment]
Weird. Is it an unstable test?

**Comment by tisonkun** [comment]
@michaeljmarshall I don't think the failures are related to this patch. Could you take a looks and rerun the tests?

**Comment by michaeljmarshall** [comment]
@tisonkun - yes, the TLS Installation and ZK & BK TLS Only tests are unstable. I have spent some time looking into them this week, but I haven't found the root cause. It seems like there may be some instability with the environment because some of the errors are about refused connections to ports that appear to be working for some pods but not others.

**Comment by michaeljmarshall** [comment]
Here is an issue for the flaky tests: #311

**Review by michaeljmarshall — APPROVED** [review]

**Pull request merged**